x-pack/plugins/maps/server/data_indexing/get_indexes_matching_pattern.ts

```diff
@@ -20,14 +21,20 @@ export async function getMatchingIndexes(
     const matchingIndexes = indexResults
       .map((indexRecord) => indexRecord.index)
       .filter((indexName) => !!indexName);
-    return {
-      success: true,
-      matchingIndexes: matchingIndexes as string[],
-    };
+    return response.ok({ body: { success: true, matchingIndexes: matchingIndexes as string[] } });
   } catch (error) {
-    return {
-      success: false,
-      error,
-    };
+    const errorStatusCode = error.meta?.statusCode;
+    if (errorStatusCode === 404) {
+      return response.ok({ body: { success: true, matchingIndexes: [] } });
+    } else {
+      logger.error(error);
```

Comment:
Do we want to surface this Error in the Kibana logs? I think since the errors are handled by the UI and messages are displayed to the user, we should not surface these errors in the kibana logs. Thoughts?

Reply:
It's consistent with the handing for our other feature editing endpoints (see [here](http://github.com/aaronjcaldwell/kibana/blob/87372c433a6f745008c24c9f7b416b6362ea4ed2/x-pack/plugins/maps/server/data_indexing/indexing_routes.ts#L102-L102) and [here](http://github.com/aaronjcaldwell/kibana/blob/87372c433a6f745008c24c9f7b416b6362ea4ed2/x-pack/plugins/maps/server/data_indexing/indexing_routes.ts#L136-L136) for example). I think there's some future work in here around possibly creating a wrapper function for endpoints where we decide if there's some standard or conditional handling we want to apply to all endpoints (error catching, logging, etc.). Since this is a bug fix, I'm mostly just trying to match existing patterns and not rock the boat too much.